### PR TITLE
Use SSE in colorbalance + clean up some color code

### PR DESCRIPTION
**Pull request by houz**
This manually merges the work done in PR #948.

### CODE REVIEW

File: src/common/sse.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/*
+    This file is part of darktable,
+    copyright (c) 2015 Bruce Guenter
```
Comment (LebedevRI) [inline src/common/sse.h]: copyright looks messed up

File: src/common/sse.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  /* ipart = int(x - 0.5) */
+  ipart = _mm_cvtps_epi32(_mm_sub_ps(x, _mm_set1_ps(0.5f)));
+
+  /* fpart = x - ipart */
+  fpart = _mm_sub_ps(x, _mm_cvtepi32_ps(ipart));
+
+  /* expipart = (float) (1 << ipart) */
+  expipart = _mm_castsi128_ps(_mm_slli_epi32(_mm_add_epi32(ipart, _mm_set1_epi32(127)), 23));
+
+/* minimax polynomial fit of 2**x, in range [-0.5, 0.5[ */
```
Comment (LebedevRI) [inline src/common/sse.h]: `[-0.5, 0.5]` probably?

File: src/iop/colorbalance.c [hunk 3]
```diff
@@ -187,6 +161,62 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
...
+      // transform the pixel to sRGB:
+      // Lab -> XYZ
+      __m128 Lab = _mm_load_ps(in);
+      __m128 XYZ = dt_Lab_to_XYZ_sse2(Lab);
+      // XYZ -> sRGB
+      __m128 rgb = dt_XYZ_to_sRGB_sse2(XYZ);
+
+      // do the calculation in RGB space
+      __m128 one = _mm_set1_ps(1.0);
+      __m128 tmp = (((rgb - one) * lift) + one) * gain;
```
Comment (LebedevRI) [inline src/iop/colorbalance.c]:
```
- __m128 tmp = (((rgb - one) * lift) + one) * gain;
+ __m128 tmp = _mm_mul_ps(_mm_add_ps(_mm_mul_ps(_mm_sub_ps(rgb, one), lift), one), gain);
```

File: src/iop/colorbalance.c [hunk 4]
```diff
@@ -187,6 +161,62 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
...
 
+#if defined(__SSE__)
+void process_sse2(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const void *const ivoid,
+             void *const ovoid, const dt_iop_roi_t *const roi_in, const dt_iop_roi_t *const roi_out)
+{
+  dt_iop_colorbalance_data_t *d = (dt_iop_colorbalance_data_t *)piece->data;
+  const int ch = piece->colors;
+
+  // these are RGB values!
+  const __m128 lift = { 2.0 - (d->lift[CHANNEL_RED] * d->lift[CHANNEL_FACTOR]),
```
Comment (LebedevRI) [inline src/iop/colorbalance.c]: I'm not 100% sure whether such initialization of `__m128` not using `_mm_set_ps()` is standard

File: src/common/sse.h (hunk 1 above)
Comment (houz) [reply src/common/sse.h]: Why is it messed up? The PR is that old ...

File: src/common/sse.h (hunk 2 above)
Comment (houz) [reply src/common/sse.h]: No idea, I didn't try to understand the code to see what ranges it's working on.

File: src/iop/colorbalance.c (hunk 3 above)
Comment (houz) [reply src/iop/colorbalance.c]: Thanks, I missed that one it seems.

File: src/iop/colorbalance.c (hunk 4 above)
Comment (houz) [reply src/iop/colorbalance.c]: And I missed that one, too. I fixed it in other places, but here I left it in. Thanks for noticing.

**Pull request merged**